test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -8283,6 +8283,146 @@ BOOST_AUTO_TEST_CASE(calldata_struct_function_type)
 	ABI_CHECK(callContractFunctionNoEncoding("f((function))", fn_C_h), encodeArgs(23));
 }
 
+BOOST_AUTO_TEST_CASE(calldata_array_dynamic)
+{
+	char const* sourceCode = R"(
+		pragma experimental ABIEncoderV2;
+		contract C {
+			function f1(uint[][1] calldata a) external returns (uint256, uint256, uint256, uint256) {
+				return (a[0].length, a[0][0], a[0][1], a[0][2]);
+			}
+			function f2(uint[][1] calldata a, uint[][1] calldata b) external returns (uint256, uint256, uint256, uint256, uint256, uint256, uint256) {
+				return (a[0].length, a[0][0], a[0][1], a[0][2], b[0].length, b[0][0], b[0][1]);
+			}
+			function g1(uint[3][2] calldata a) external returns (uint256, uint256, uint256, uint256, uint256, uint256) {
+				return (a[0][0], a[0][1], a[0][2], a[1][0], a[1][1], a[1][2]);
+			}
+			function g2(uint[][2] calldata a) external returns (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256) {
+				return (a[0].length, a[0][0], a[0][1], a[0][2], a[1].length, a[1][0], a[1][1], a[1][2]);
+			}
+			function g3(uint[][] calldata a) external returns (uint256[8] memory) {
+				return [a.length, a[0].length, a[0][0], a[0][1], a[1].length, a[1][0], a[1][1], a[1][2]];
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+
+	ABI_CHECK(
+	    callContractFunction("f1(uint256[][1])", 0x20, 0x20, 3, 1, 2, 3),
+	    encodeArgs(3, 1, 2, 3)
+	);
+	ABI_CHECK(
+	    callContractFunction("f2(uint256[][1],uint256[][1])", 0x40, 0xE0, 0x20, 3, 1, 2, 3, 0x20, 2, 1, 2),
```

Comment:
```suggestion
	    callContractFunction("f2(uint256[][1],uint256[][1])", 0x40, 0xE0, 0x20, 3, 1, 2, 3, 0x20, 2, 7, 8),
```
Better to not repeat any values.

Actually it would be even better if the length would not equal one of the element values.